module/VuFind/src/VuFind/Mailer/Mailer.php

```diff
@@ -487,21 +437,26 @@ public function setFromAddressOverride($address)
     }
 
     /**
-     * Convert the given addresses to an AddressList object
+     * Convert the given addresses to an array
      *
-     * @param string|Address|AddressList $addresses Addresses
+     * @param string|Address|Address[]|null $addresses Addresses
      *
-     * @return AddressList
+     * @return array
      */
-    protected function convertToAddressList($addresses)
+    protected function convertToAddressList(string|Address|array|null $addresses): array
     {
-        if ($addresses instanceof AddressList) {
-            $result = $addresses;
-        } elseif ($addresses instanceof Address) {
-            $result = new AddressList();
-            $result->add($addresses);
-        } else {
-            $result = $this->stringToAddressList($addresses ? $addresses : '');
+        if (empty($addresses)) {
+            return [];
+        }
+        if ($addresses instanceof Address) {
+            return [$addresses];
+        }
+        if (is_array($addresses)) {
+            return Address::createArray($addresses);
```

Comment:
Does this account for the cases of $addresses being an array of strings OR an array of Address objects? Would it make sense to add an `array_map(fn ($addr) => $addr instanceof Address ? $addr : new Address($addr), $addresses)` to normalize if not, just to support more potential scenarios?

Reply:
It does. I added a comment.